googlesearch/__init__.py

```diff
@@ -256,10 +250,9 @@ def search(query, tld='com', lang='en', tbs='0', safe='off', num=10, start=0,
     # If no extra_params is given, create an empty dictionary.
     # We should avoid using an empty dictionary as a default value
     # in a function parameter in Python.
-    if not extra_params:
-        extra_params = {}
+    extra_params = {} if not extra_params else extra_params
 
-    # Check extra_params for overlapping.
+    # Check extra_params for overlapping. # <== how necessary is this? Especially if it's a set?
```

Comment:
This is to prevent logic bugs when calling the library - if someone specifies the same parameter twice, once via the extra parameters dictionary and another via the regular parameters